SRM 325/RGBStreet.cpp

```diff
@@ -0,0 +1,209 @@
+#include <cstdio>
+#include <cmath>
+#include <cstring>
+#include <ctime>
+#include <iostream>
+#include <algorithm>
+#include <set>
+#include <vector>
+#include <sstream>
+#include <typeinfo>
+#include <fstream>
+
+using namespace std;
+
+int dp[100][4];
+
+class RGBStreet {
+    public:
+
+    int convertStringToInt(string s) {
+        int numb = 0;
+        for(int i=0; i<s.length(); ++i) {
+            numb = numb*10 + (s[i]-'0');
+        }
+        return numb;
+    }
+
+    vector<vector<int> > extractPrices(vector<string> houses) {
+        vector<vector<int> > prices(houses.size());
+        for(int i=0 ;i<houses.size(); ++i) {
+            int start = 0;
+            vector<int> tmp;
+            for(int j=0; j<houses[i].length(); ++j) {
+                if(houses[i][j] == ' ') {
+                    string str1 = houses[i].substr(start, j-start);
+                    tmp.push_back(convertStringToInt(str1));
+                    start = j+1;  
+                }
+                else if(j+1 == houses[i].length()) {
+                    string str1 = houses[i].substr(start);
+                    tmp.push_back(convertStringToInt(str1));
+                }
+            }
+            prices[i] = tmp;
+        }
+        return prices;
+    }
+
+    int f(int i, int previousHousePaint, vector<string>& houses, vector< vector< int> >& prices) {
+        int &res = dp[i][previousHousePaint];
+        if(res == -1) {
+            if(i >= houses.size()) {
+                res = 0;
+            }
+            else{
+                if(previousHousePaint == -1) {
```

Comment:
You can combine this case with below. Just recur on the subproblem for the cases where `previousHousePaint ` is different. Like 
```
if(previousHousePaint != 0) {
    // use color 0 at this position
}

if(previousHousePaint != 1) {
    // use color 1 at this position
}

if(previousHousePaint != 2) {
    // use color 2 at this position
}
```